### PR TITLE
Wiremock random state improvement

### PR DESCRIPTION
**Pull request by Mateusz512**
## Description
Wiremock responses now return random state with each request.

## Motivation and Context

It's easier to demonstrate caching behaviour if the external API is returning different response with each request.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the [code style](/CONTRIBUTING.md#coding-conventions) of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

---
I hereby agree to the terms of the Knot.x Contributor License Agreement.

### CODE REVIEW

File: api-gateway/caching-api-response/api-cache.yml [hunk 1]
```diff
@@ -9,10 +9,13 @@ services:
     image: rodolpheche/wiremock
     volumes:
       - "../../common-services/webapi:/home/wiremock"
+      # Remember to put https://repo1.maven.org/maven2/com/opentable/wiremock-body-transformer/1.1.3/wiremock-body-transformer-1.1.3.jar in ../../common-services/webapi/extensions directory!
```
Comment (tomaszmichalak) [inline api-gateway/caching-api-response/api-cache.yml]: Can we automate it with Gradle? This jar file should be ignored by Git.

File: common-services/webapi/__files/product.json [hunk 2]
```diff
@@ -1,5 +1,7 @@
 {
   "id": 21762532,
   "url": "http://knotx.io",
-  "label": "Product"
+  "label": "Product",
+  "currentTime": "{{now format='yyyy-MM-dd HH:mm:ss'}}",
```
Comment (tomaszmichalak) [inline common-services/webapi/__files/product.json]: Nice!

File: common-services/webapi/__files/product.json [hunk 3]
```diff
@@ -1,5 +1,7 @@
 {
   "id": 21762532,
   "url": "http://knotx.io",
-  "label": "Product"
+  "label": "Product",
```
Comment (tomaszmichalak) [inline common-services/webapi/__files/product.json]: Please provide some information in `./README.md` about the target API responses (what is changing) and then explain API Gateway responses (nothing is changed until cached value is valid).

File: api-gateway/caching-api-response/api-cache.yml (hunk 1 above)
Comment (Mateusz512) [reply api-gateway/caching-api-response/api-cache.yml]: done

File: common-services/webapi/__files/product.json (hunk 2 above)
Comment (Mateusz512) [reply common-services/webapi/__files/product.json]: 🍻

File: common-services/webapi/__files/product.json (hunk 3 above)
Comment (Mateusz512) [reply common-services/webapi/__files/product.json]: done

File: api-gateway/caching-api-response/build.gradle.kts [hunk 4]
```diff
@@ -19,8 +19,13 @@ plugins {
     id("java")
 }
 
+configurations {
+    register("wiremockExtensions")
+}
+
 dependencies {
     subprojects.forEach { "dist"(project(":${it.name}")) }
+    "wiremockExtensions"("com.opentable:wiremock-body-transformer:1.1.3") { isTransitive = false }
```
Comment (tomaszmichalak) [inline api-gateway/caching-api-response/build.gradle.kts]: Super!

**Pull request merged**